client/structs/structs.go

```diff
@@ -183,6 +183,10 @@ type AllocExecRequest struct {
 	Cmd []string
 
 	structs.QueryOptions
+
+	// The name of a predefined command to be executed (optional)
+	Action string
+	JobID  string
```

Comment:
Style nitpick so that the embedded struct falls at the end of the list of fields (we typically have it there or in the beginning of the list). Also, do we still need `JobID` in this struct?

```suggestion
	// The name of a predefined command to be executed (optional)
	Action string
	JobID  string

	structs.QueryOptions
```